src/vmm/src/arch/aarch64/regs.rs

```diff
@@ -115,6 +115,9 @@ arm64_sys_reg!(KVM_REG_ARM_TIMER_CNT, 3, 3, 14, 3, 2);
 pub const KVM_REG_ARM64_SVE_VLS: u64 =
     KVM_REG_ARM64 | KVM_REG_ARM64_SVE as u64 | KVM_REG_SIZE_U512 | 0xffff;
 
+/// Program Counter
+pub const PC: u64 = arm64_core_reg_id!(KVM_REG_SIZE_U64, 256);
```

Comment:
I think keeping the calculation of `PC` is valuable (otherwise where does 256 comes from). Also maybe worth removing the custom `offset_of` macro from `regs.rs` as there is same macro in std. You can do:
```rust
use std::mem::offest_of;
pub const PC: u64 = {
  let kreg_off = offset_of!(kvm_regs, regs);
  let pc_off = offset_of!(user_pt_regs, pc);
  arm64_core_reg_id!(KVM_REG_SIZE_U64, kreg_off + pc_off)
};
```

Reply:
`offset_of!()` seems to be an unstable feature.
```
error[E0658]: use of unstable library feature 'offset_of'
   --> src/vmm/src/arch/aarch64/regs.rs:122:20
    |
122 |     let kreg_off = offset_of!(kvm_regs, regs);
    |                    ^^^^^^^^^
    |
    = note: see issue #106655 <https://github.com/rust-lang/rust/issues/106655> for more information
```

To avoid this, we have our own `offset__of!()`, but it doesn't work during const eval at compile time :(
```
error: pointers cannot be cast to integers during const eval
   --> src/vmm/src/arch/aarch64/regs.rs:35:13
    |
35  |             std::ptr::addr_of!((*ptr).$field) as usize - ptr as usize
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
...
120 |     let kreg_off = offset__of!(kvm_regs, regs);
    |                    --------------------------- in this macro invocation
    |
    = note: at compile-time, pointers do not have an integer value
    = note: avoiding this restriction via `transmute`, `union`, or raw pointers leads to compile-time undefined behavior
    = note: this error originates in the macro `offset__of` (in Nightly builds, run with -Z macro-backtrace for more info)

```

Reply:
I can add a comment where it comes from :)

https://docs.kernel.org/virt/kvm/api.html
> | 0x6030 0000 0010 0040 | PC | 64 | regs.pc |

Reply:
It seems it becomes stable since 1.77.0: https://doc.rust-lang.org/nightly/src/core/mem/mod.rs.html#1398-1401

Reply:
I'll add TODO comment there.